flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/type/RecordValueGenerator.java

```diff
@@ -0,0 +1,212 @@
+            for (JsonElement member : members.getAsJsonArray()) {
+                JsonObject memberObj = member.getAsJsonObject();
+                if (memberObj.has("selected") && memberObj.get("selected").getAsBoolean()) {
+                    if (memberObj.has("value") && !memberObj.get("value").getAsString().isEmpty()) {
+                        builder.append(memberObj.get("value").getAsString());
+                    } else if (memberObj.has("defaultValue") &&
+                            !memberObj.get("defaultValue").getAsString().isEmpty()) {
+                        builder.append(memberObj.get("defaultValue").getAsString());
+                    } else {
+                        StringBuilder sb = new StringBuilder();
+                        generateDefaultValue(memberObj, sb);
+                        builder.append(sb);
+                    }
+                    break;
+                }
+            }
+        }
+    }
+
+    public static void generateUnionValue(JsonObject union, StringBuilder builder) {
+        if (union.has("members")) {
+            JsonElement members = union.get("members");
+            if (members.isJsonArray()) {
+                for (JsonElement member : members.getAsJsonArray()) {
+                    JsonObject memberObj = member.getAsJsonObject();
+                    if (memberObj.has("selected") && memberObj.get("selected").getAsBoolean()) {
+                        if (memberObj.has("typeName")) {
+                            String typeName = memberObj.get("typeName").getAsString();
+                            switch (typeName) {
+                                case "record" -> {
+                                    generateRecordValue(memberObj, builder);
+                                }
+                                case "union" -> {
+                                    generateUnionValue(memberObj, builder);
+                                } default -> {
+                                    if (memberObj.has("value") &&
+                                            !memberObj.get("value").getAsString().isEmpty()) {
+                                        builder.append(memberObj.get("value").getAsString());
+                                    } else if (memberObj.has("defaultValue")
+                                            && !memberObj.get("defaultValue").getAsString().isEmpty()) {
+                                        builder.append(memberObj.get("defaultValue").getAsString());
+                                    } else {
+                                        StringBuilder sb = new StringBuilder();
+                                        generateDefaultValue(memberObj, sb);
+                                        builder.append(sb);
+                                    }
+                                }
+                            }
+                        }
+                        break;
```

Comment:
Can we guarantee that one is selected everytime?

Reply:
Yes, we can. We have a selection drop down to select the member type of the union.

https://github.com/user-attachments/assets/bb04791a-cfe0-4610-90e4-97074d61c7f5